### ISSUE TITLE
normalize(norm="max") divides by the signed maximum, not the peak absolute value

### ISSUE THREAD

**Issue by d-chambers** [DASDAE/dasjax#2]
`normalize_kernel` divides the `max` case by `jnp.max(data, axis=axis)`. DASCore divides by `np.max(np.abs(data), axis=axis)`, and has since `normalize` was added in DASDAE/dascore#52 — the docstring states it plainly:

> max - divide each sample by the maximum of the absolute value of the axis.

So for any slice whose most negative sample is larger in magnitude than its most positive one, the two disagree in both scale and sign, and the result is no longer bounded by 1. This is not drift against `dev`: released DASCore (v0.1.20) uses the absolute value too.

Both execution paths are affected, since the eager `normalize_patch` and the compiled `normalize_leaves` share the kernel.

The comment above the branch asserts the opposite of what DASCore does, which is probably where it came from:

https://github.com/DASDAE/dasjax/blob/main/src/dasjax/kernels/signal.py#L67-L71

```python
if norm == "max":
    # Max normalization matches DASCore's sign-preserving divide-by-peak behavior.
    norm_values = jnp.max(data, axis=axis)
```

## Reproducing

`main`'s own parity suite already catches this — the declared `normalize-max` case is compared against `Patch.normalize`, and it fails on four of the ten fixture patches, in both the eager and compiled tests:

```
$ pytest tests/test_method_equivalence.py -k normalize -q
FAILED ...test_operation_patch_impl_matches_declared_baseline[example_event_1-normalize-normalize-max]
FAILED ...test_operation_patch_impl_matches_declared_baseline[deformation_rate_event_1-normalize-normalize-max]
FAILED ...test_operation_patch_impl_matches_declared_baseline[nd_patch-normalize-normalize-max]
FAILED ...test_operation_patch_impl_matches_declared_baseline[patch_with_null-normalize-normalize-max]
FAILED ...test_compiled_pipeline_matches_declared_baseline[example_event_1-normalize-normalize-max]
FAILED ...test_compiled_pipeline_matches_declared_baseline[deformation_rate_event_1-normalize-normalize-max]
FAILED ...test_compiled_pipeline_matches_declared_baseline[nd_patch-normalize-normalize-max]
FAILED ...test_compiled_pipeline_matches_declared_baseline[patch_with_null-normalize-normalize-max]
```

The six fixtures that pass all happen to satisfy `max(x) == max(|x|)`, which is why the case looks covered.

Standalone, on real DAS:

```python
import numpy as np
import dascore as dc
from dasjax import JaxPatchPipeline

patch = dc.get_example_patch("example_event_1")
a = patch.normalize("time", norm="max").data
b = JaxPatchPipeline().normalize(dim="time", norm="max").compile()(patch).data
print(np.abs(a - b).max())   # 0.626, on data DASCore bounds to [-1, 1]
```

## Suggested fix

```python
norm_values = jnp.max(jnp.abs(data), axis=axis)
```

## Related, same function

The same run fails `normalize-l1` and `normalize-l2` on `patch_with_null`. That one *is* drift against `dev`: DASCore's `normalize` now skips NaN when computing the norm, so a single null no longer blanks every sample sharing its slice, while `jnp.linalg.norm` still propagates it. Worth folding into the same fix, or splitting out if you would rather keep them separate.

## Where this came from

Writing the DASJax notebook for the [EGU Galileo tutorial](https://github.com/DASDAE/galileo_2026), which builds a compiled pipeline and asserts it agrees with the equivalent DASCore chain. `norm="max"` was the natural choice there — it is the one that makes per-channel equalization obvious — but it disagreed by 0.95 on data ranging [-1, 1], so the notebook uses `l2` for now.

Tested against dasjax `main` (4df967b) and 0.0.2, with DASCore `dev`.
